addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlBridgeHandler.java

```diff
@@ -0,0 +1,198 @@
+
+        Configuration config = this.getConfig();
+        // workaround for int port not being read back as int from configuration, remove trailing .0
+        logger.debug("Niko Home Control: bridge handler port {}", config.get(CONFIG_PORT).toString());
+        String portString = config.get(CONFIG_PORT).toString();
+        int portStringDotPos = portString.indexOf(".");
+        if (portStringDotPos > 0) {
+            portString = portString.substring(0, portStringDotPos);
+        }
+        port = Integer.parseInt(portString);
+        try {
+            if (config.get(CONFIG_HOST_NAME) != null) {
+                logger.debug("Niko Home Control: bridge handler host {}", config.get(CONFIG_HOST_NAME).toString());
+                addr = InetAddress.getByName(config.get(CONFIG_HOST_NAME).toString());
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Niko Home Control: cannot resolve bridge IP");
+            return;
+        }
+        BigDecimal intervalConfig = (BigDecimal) config.get(CONFIG_REFRESH);
+        int refreshInterval;
+        if (intervalConfig == null) {
+            refreshInterval = 0;
+        } else {
+            refreshInterval = intervalConfig.intValue();
+        }
+
+        InetAddress nhcAddr = addr;
+
+        createCommunicationObject(nhcAddr, port);
+        setupRefreshTimer(refreshInterval);
+
+    }
+
+    /**
+     * Create communication object to Niko Home Control gateway and start communication.
+     * Trigger discovery when communication setup is successful.
+     *
+     * @param nhcAddr IP address or subnet for gateway
+     * @param port
+     */
+    private void createCommunicationObject(InetAddress nhcAddr, int port) {
+
+        ThingDiscoveryService nhcDiscovery = new ThingDiscoveryService(thing.getUID(), this);
+
+        scheduler.submit(new Runnable() {
+
+            @Override
+            public void run() {
+                try {
+                    for (int i = 0; i < 3; i++) {
+                        // try connecting max 3 times
+                        nhcComm = new NikoHomeControlCommunication(port, nhcAddr);
+                        if (nhcComm.communicationActive()) {
+                            break;
+                        }
+                        nhcComm = null;
+                    }
+                    if (!nhcComm.communicationActive()) {
+                        IOException e = new IOException("Niko Home Control: communication socket not open");
+                        throw e;
+                    }
+                    thing.getConfiguration().put(CONFIG_HOST_NAME, nhcComm.getAddr().getHostAddress());
+                    thing.getConfiguration().put(CONFIG_PORT, port);
+                    HashMap<String, String> properties = new HashMap<String, String>();
+                    properties.put("IP Address", nhcComm.getAddr().getHostAddress());
```

Comment:
I was in doubt, but followed your suggestion. I also put this all in a separate method, so I can easily call it on restart of the communication as well.